src/stan/services/init/initialize_state.hpp

```diff
@@ -290,34 +286,36 @@ namespace stan {
             std::vector<double> cont_vecs_constrained;
             std::vector<int> int_vecs;
             model.write_array(base_rng, cont_vecs, int_vecs,
-                              cont_vecs_constrained, false, false, output);
+                              cont_vecs_constrained, false, false, 0);
             std::vector<std::vector<size_t> > dims;
             model.get_dims(dims);
             stan::io::array_var_context random_context(cont_names,
                                                        cont_vecs_constrained,
                                                        dims);
             stan::io::chained_var_context cvc(context, random_context);
-            model.transform_inits(cvc, cont_params, output);
-            if (initialize_state_values(cont_params, model, output))
+            model.transform_inits(cvc, cont_params, 0);
```

Comment:
This stuff will all change with the var_context refactor so it may not matter for long.  We can remove the output or use local string streams to capture any error output and pass them to the writer.